src/main/java/net/datafaker/idnumbers/SwedenIdNumber.java

```diff
@@ -37,14 +37,23 @@ public String getValidSsn(BaseProviders f) {
     @Override
     public PersonIdNumber generateValid(BaseProviders f, IdNumber.IdNumberRequest request) {
         LocalDate birthday = birthday(f, request);
-        String end = f.numerify("###");
+        String end = generateEndPart(f);
         String basePart = DATE_TIME_FORMATTER.format(birthday)
             + f.options().option(PLUS_MINUS)
             + end;
         String idNumber = basePart + calculateChecksum(basePart);
         return new PersonIdNumber(idNumber, birthday, gender(f, request));
     }
 
+    private String generateEndPart(BaseProviders f) {
+        String end;
+        do {
+            end = f.numerify("###");
+        } while (!end.equals("000"));
```

Comment:
Isn't it a bug?
Shouldn't it be the opposite: `while (end.equals("000"));`?

Reply:
Ha, good catch. I think you're right

Reply:
Yeah, you're right 😅. Already fixed